[DEVECO-1007] Rework Build Apps & Integrations; document Public Scoped Apps (EA) (#144)
PagerDuty/developer-docs: docs/app-integration-development/01-Overview.md

#### What is an app?

> An app allows customers to set up other web-enabled businesses to work with PagerDuty or to make PagerDuty more useful. Apps can send data to PagerDuty or receive data from PagerDuty.

Reviewer: ```suggestion
An app allows customers to set up other web-enabled businesses to work with PagerDuty or to make PagerDuty more useful. Apps can send data to PagerDuty, receive data from PagerDuty, or both.
```

#### Who builds apps?
Apps are built by PagerDuty, our partners, our customers, system integrators, and the open source community! Any PagerDuty customer or developer with a [PagerDuty developer account](03-Developer-Account.md) has access to PagerDuty app development tools.

#### Can my app be used by other PagerDuty accounts?

> Yes, apps can be created for either Private use (on your account only) or Public use which can be installed and used by other PagerDuty customers. See [Private Apps](02-Private-Apps.md) and [Publish an App](08-Publish-Your-App.md).

Reviewer: ```suggestion
Yes, apps can be created for either _private use_ (on your account only) or _public use_ which can be installed and used by other PagerDuty customers. See [Private Apps](02-Private-Apps.md) and [Publish an App](08-Publish-Your-App.md).
```

### Getting Started
1. Not a PagerDuty customer? [Sign up for a developer account to get access to PagerDuty](https://developer.pagerduty.com/sign-up/).
1. [Register your app and add functionality to it](04-Register-an-App.md)
1. [Publish an app (completely optional)](08-Publish-Your-App.md)

## App Functionality

> There are two kinds of app functionality. An app can have either, or both.

Reviewer: nit: It feels a little strange to say "There are two kinds of app functionality" but not state what they are...

### Events Integration

> [Events Integration](05-Events-Integration.md) sends machine events **from** your tool **to** PagerDuty over the asynchronous [Events API v2](../../docs/events-API-v2/01-Overview.md). This is the best way for a monitoring tool to connect with PagerDuty, and it can trigger, acknowledge, and resolve incidents.

Reviewer: ```suggestion
[Events Integration](05-Events-Integration.md) sends machine events **from** your tool **to** PagerDuty via the [v2 Events API](../../docs/events-API-v2/01-Overview.md). This is the best way for a monitoring tool to connect with PagerDuty, and it can trigger, acknowledge, and resolve incidents.
```

The Events API is asynchronous and designed for high volume. Events flow through PagerDuty's [Event Intelligence](https://www.pagerduty.com/platform/event-intelligence-and-automation/) features, which group, deduplicate, and triage them to reduce noise and speed up response.

> It can also send [change events](../../docs/events-API-v2/03-Change-Events.md) — informational signals about recent changes such as code deploys and system config changes, which are displayed in PagerDuty rather than triggering an incident.

Reviewer: ```suggestion
It can also send [Change Events](../../docs/events-API-v2/03-Change-Events.md) — informational signals about recent changes such as code deploys and system config changes, which are displayed in PagerDuty rather than triggering an incident.
```

Reviewer: (If we are capitalizing our resource names)

Two optional pieces come with it:

> * The [Simple Install Flow](05-Events-Integration.md#simple-install-flow-optional-but-recommended) lets your users connect to PagerDuty from inside your tool, instead of copying and pasting integration keys. [See Demo](https://acme.pagerduty.dev)

Reviewer: ```suggestion
* The [Simple Install Flow](05-Events-Integration.md#simple-install-flow-optional-but-recommended) lets your users connect to PagerDuty from inside your tool, instead of copying and pasting integration keys. See the demo [here](https://acme.pagerduty.dev).
```
